istioctl/pkg/waypoint/waypoint.go

```diff
@@ -446,15 +460,26 @@ func labelNamespaceWithWaypoint(kubeClient kube.CLIClient, ns string) error {
 	return nil
 }
 
-func namespaceIsLabeledAmbient(kubeClient kube.CLIClient, ns string) (bool, error) {
-	nsObj, err := kubeClient.Kube().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		return false, fmt.Errorf("namespace: %s not found", ns)
-	} else if err != nil {
-		return false, fmt.Errorf("failed to get namespace %s: %v", ns, err)
+func namespaceHasLabel(kubeClient kube.CLIClient, ns string, label string) (bool, error) {
+	nsObj, err := getNamespace(kubeClient, ns)
+	if err != nil {
+		return false, err
 	}
 	if nsObj.Labels == nil {
 		return false, nil
 	}
-	return nsObj.Labels[constants.DataplaneModeLabel] == constants.DataplaneModeAmbient, nil
+	if label == constants.DataplaneModeLabel {
```

Comment:
Might as well generalize this - if we want to distinguish between `hasLabel` and `hasLabelWithValue` lets do that explicitly, and not special-case the ambient dataplane label inside a function that sounds generic/has a generic signature.

Reply:
> hasLabel and hasLabelWithValue

I played around with this in my head for a moment but couldn't think of a way to easily generalize for some reason. `hasLabelWithValue` is so obvious but clean, good idea. Will update!